ticketbuyer/purchase.go

```diff
@@ -81,11 +82,15 @@ out:
 				case <-quit:
 					return
 				}
-				// Purchase tickets for each attached block, not just for the
-				// update to the main chain.  This is probably not optimal but
-				// it matches how dcrticketbuyer worked.
-				for h := v.NewHeight - int32(len(v.AttachedBlocks)) + 1; h <= v.NewHeight; h++ {
-					p.purchase(int64(h))
+
+				blockInfo, err := p.w.BlockInfo(wallet.NewBlockIdentifierFromHeight(v.NewHeight))
+				if err != nil {
+					log.Errorf("failed to get block info for bock height %d", v.NewHeight)
```

Comment:
misspelled block

Reply:
also this should be using the block hash of the most recently attached block, not the height